main/dns-root-hints/APKBUILD

```diff
@@ -1,39 +1,69 @@
 # Contributor: Leonardo Arena <rnalrd@alpinelinux.org>
 # Contributor: tcely <dnshints-root+aports@tcely.33mail.com>
 # Maintainer: tcely <dnshints-root+aports@tcely.33mail.com>
-pkgname=dns-root-hints
-pkgver=2019031302
-pkgrel=1
+pkgname="dns-root-hints"
+pkgver="2019073000"
+pkgrel=0
 pkgdesc="The DNS root hint(s)"
 url="https://www.internic.net/domain"
 arch="noarch"
 license="Public-Domain"
 makedepends="curl gnupg"
 options="net"
-source="verisign-grs-nstld-key.asc named.root named.root.sig"
-builddir="$srcdir/$pkgname-$pkgver"
+source="
+	${pkgname}-${pkgver}.tar.gz::https://github.com/tcely/internic-domain-data/archive/${pkgver}.tar.gz
+	verisign-grs-nstld-key.asc
+	"
+builddir="${srcdir}/internic-domain-data-${pkgver}"
 
-build() {
-	mkdir -p "$builddir"
+prepare() {
+	default_prepare
+
+	builddir="${builddir}/build"
+	mkdir -p "${builddir}"
 }
 
-package() {
-	cd "$builddir"
-	install -D -m 644 -o root -g root "$srcdir"/named.root \
-		"$pkgdir"/usr/share/$pkgname/named.root
-	install -D -m 644 -o root -g root "$srcdir"/named.root.sig \
-		"$pkgdir"/usr/share/$pkgname/named.root.sig
-	install -D -m 644 -o root -g root "$srcdir"/verisign-grs-nstld-key.asc \
-		"$pkgdir"/usr/share/$pkgname/verisign-grs-nstld-key.asc
+_check_sig() {
+	local GNUPGHOME="${builddir}/.gpg"
+	install -d -m 0700 "${GNUPGHOME}"
+	gpg --import < "${srcdir}/verisign-grs-nstld-key.asc"
+	gpg --verify "${_tmp}/named.root.sig" "${_tmp}/named.root"
+	rm -rf "${GNUPGHOME}"
+	mv "${_tmp}/named.root.sig" "${_tmp}/named.root" "${builddir}/"
+}
 
-	# compatibility links
-	cd "$pkgdir/usr/share/$pkgname"
-	ln -s named.root named.cache
-	ln -s named.root db.cache
+_copy_from_source_archive() {
+	cp -p "${builddir}"/../domain/named.root* .
+}
+
+_fetch_from_url() {
+	curl -sSRL -z ./named.root -O "${url}/named.root{.sig,}"
+}
+
+_gather_named_root_files() {
+	local _tmp="$(mktemp -d -p .)"
+	cd "${_tmp}"
+	_copy_from_source_archive
+	options_has net && _fetch_from_url
+	cd -
+	_check_sig
+	rm -rf "${_tmp}"
+}
+
+build() {
+	cd "${builddir}"
+	test ! -d build || cd build
```

Comment:
Why would build not exist at this point? It has been created in prepare

Reply:
The value of `builddir` depends on which functions have already run. When `/build` was not appended the build subdirectory exists and we cd into it.

Reply:
You did point me at a problem I didn't handle. The value of `builddir` was not accurate for other functions when the `cd` case hit. bc4a0cc corrects that problem.